tests/test_modeling_common.py

```diff
@@ -4355,6 +4393,32 @@ def test_custom_4d_attention_mask(self):
             normalized_1 = F.softmax(out_shared_prefix_last_tokens)
             torch.testing.assert_close(normalized_0, normalized_1, rtol=1e-3, atol=1e-4)
 
+    # For now, Let's focus only on GPU for `torch.compile`
+    @slow
+    @require_torch_gpu
+    @require_read_token
+    def test_torch_compile(self):
+        if version.parse(torch.__version__) < version.parse("2.2.1"):
+            self.skipTest("This test requires torch >= 2.2.1 to run.")
+
+        if not hasattr(self, "_torch_compile_test_ckpt"):
+            self.skipTest(f"{self.__class__.__name__} doesn't have the attribute `_torch_compile_test_ckpt`.")
+        ckpt = self._torch_compile_test_ckpt()
+
+        with tempfile.TemporaryDirectory() as tmpdirname:
```

Comment:
why do we do this with multiprocessing?

Reply:
You don't read my great comment https://github.com/huggingface/transformers/commit/a63c3ba1f21b8f3400b24a87a9d8607d489edb2a 😭 

Here it is
> The actual logic, which should be run inside a subprocess (called in _test_torch_compile).
The reason for this is that, running torch.compile multiple times in the same process will accumulate the GPU usage and hard (if not impossible) to clean

Reply:
It's probably not really necessary as we split our tests by models in our CI.

Reply:
You can use 
```
# Reset since we are using a different mode.
import torch._dynamo
torch._dynamo.reset()
```
no?

Reply:
but if not ok got it!

Reply:
sound a much better solution. Thanks a lot Sir! 

I will check it

Reply:
Unfortunately, I don't see any effect (via `nvidia-smi`) on the memory usage reduce by using `torch._dynamo.reset()`.

Reply:
I will remove the usage of subprocess howerver as since each model is run in an independent CI job, the memory issue won't actually happen